File: glug-gui-components/src/test/java/glug/gui/SwingHtmlStyleThreadReporterTest.java
package glug.gui;

import glug.model.ThreadModel;
import glug.model.ThreadedSystem;
import glug.model.time.LogInterval;
import org.junit.Before;
import org.junit.Test;

import static java.awt.Color.RED;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
import static org.joda.time.Duration.standardSeconds;


public class SwingHtmlStyleThreadReporterTest {

    private SwingHtmlStyleThreadReporter reporter;
    private ThreadedSystem threadedSystem;

    @Before
    public void setUp() {
        reporter = new SwingHtmlStyleThreadReporter();
        threadedSystem = new ThreadedSystem();
    }

    @Test
    public void shouldCopeWithAbsentIntervalTypesForRequestedInstant() {
        ThreadModel thread = threadedSystem.getOrCreateThread("my-thread");
        reporter.htmlSyledReportFor(thread, LogInstant.apply(1000));
    }


    @Test
    public void shouldIncludeUptimeIfKnown() {
        threadedSystem.uptime().addUptime(new LogInterval(standardSeconds(3), LogInstant.apply(3000)));
        assertThat(reporter.uptimeStringFor(threadedSystem, LogInstant.apply(1000)), containsString("uptime: 1.000 s"));
    }

    @Test
    public void shouldNotThrowExceptionIfUptimeIsNotKnown() {
        assertThat(reporter.uptimeStringFor(threadedSystem, LogInstant.apply(1000)), equalTo(""));
    }

    @Test
    public void shouldReturnRightHexForColour() {
        assertThat(reporter.hexFor(RED), equalToIgnoringCase("FF0000"));
    }
}

File: glug-gui-components/src/test/scala/glug/gui/SwingHtmlStyleThreadReporterTest.scala
package glug.gui

import java.awt.Color.RED

import glug.model.ThreadedSystem
import glug.model.time.{LogInstant, LogInterval}
import org.joda.time.Duration.standardSeconds
import org.scalatest.{FlatSpec, Matchers}
import org.scalactic.StringNormalizations._

class SwingHtmlStyleThreadReporterTest extends FlatSpec with Matchers {

  val reporter = new SwingHtmlStyleThreadReporter

  "reporter" should "cope with absent IntervalType for requested instant" in {
    val threadedSystem = new ThreadedSystem
    val thread = threadedSystem.getOrCreateThread("my-thread")
    reporter.htmlSyledReportFor(thread, new LogInstant(1000, 0))
  }

  it should "include uptime if known" in {
    val threadedSystem = new ThreadedSystem
    threadedSystem.uptime.addUptime(LogInterval.apply(standardSeconds(3), new LogInstant(3000, 0)))
    reporter.uptimeStringFor(threadedSystem, LogInstant.apply(1000)) should include("uptime: 1.000 s")
  }

  it should "NotThrowExceptionIfUptimeIsNotKnown" in {
    val threadedSystem = new ThreadedSystem
    reporter.uptimeStringFor(threadedSystem, LogInstant.apply(1000)) shouldBe ""
  }

  it should "shouldReturnRightHexForColour" in {
    reporter.hexFor(RED) should equal("ff0000") (after being lowerCased)
  }
}